prisma/migrations/20220712022829_init/migration.sql

```diff
@@ -0,0 +1,21 @@
+/*
+  Warnings:
+
+  - A unique constraint covering the columns `[id,subject,code,year,term]` on the table `Course` will be added. If there are existing duplicate values, this will fail.
+  - Added the required column `courseCode` to the `Preference` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `courseSubject` to the `Preference` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `courseTerm` to the `Preference` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `courseYear` to the `Preference` table without a default value. This is not possible if the table is not empty.
+
+*/
+-- AlterTable
+ALTER TABLE "Preference" ADD COLUMN     "courseCode" TEXT NOT NULL,
```

Comment:
Can you explain this relation to me? Does this mean that prof preferences are grouped by term/year? So if a prof wants to teach CSC 230 which is offered fall/spring, they'd have a preference for 
```
Code: 230
Subject: CSC
Term: FALL
Year: 2022
```
and
```
Code: 230
Subject: CSC
Term: SPRING
Year: 2022
```
?

Reply:
Based on our shared schema https://github.com/SENG499-S22-Company3/shared/blob/main/graphql/schema.graphql#L130

Would it make more sense to just have it based on a year rather then the pair of year and term? I can bring it up in the company3/4 channel